model-optimizer/extensions/back/ReverseInputChannels.py

```diff
@@ -94,8 +94,36 @@ class ReverseChannelsPropagationDown(BackReplacementPattern):
 
         'Shape': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through_shape(node, rc),
         'ShapeOf': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through_shape(node, rc),
+
+        'Pad': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through(node, rc),
     }
 
+    @staticmethod
+    def pass_rc_through(node: Node, reverse_channels: Node):
+        r"""
+        BEFORE                          AFTER
+
+          previous_op
+              |
+        ReverseChannels begin end       previous_op begin end
+                     \    /   /                   \   /  /
+                        Pad                         Pad
+                                                     |
+                                              ReverseChannels
+
+        returns boolean value whatever we should continue propagating current ReverseChannels operation down or not
+        """
+        # detaching reverse_channels node from the graph
+        reverse_channels.out_port(0).get_connection().set_source(
+            reverse_channels.in_port(0).get_connection().get_source())
+        reverse_channels.in_port(0).disconnect()
```

Comment:
also, please check that all needed input/output ports are connected using `assert`